contributing/building-a-bento-amp-extension.md

```diff
@@ -0,0 +1,701 @@
+upon a Preact component, but this is not a requirement. The only requirement of a Bento component is that it can be fully functional on non-AMP pages. However, when there is an underlying Preact component, that component must also be usable in isolation in React development contexts.
+
+If you'd like to add an extension to support your company
+video player, rich embed or just a general UI component like a star
+rating viewer, you'd do this by building an extension. Alternatively, if
+you'd like to upgrade an existing AMP extension to be compatible with
+non-AMP pages, you'd do this by building an extension in the Bento paradigm.
+
+-   [Getting started](#getting-started)
+-   [Naming](#naming)
+-   [Directory structure](#directory-structure)
+-   [Extend AMP.PreactBaseElement](#extend-amppreactbaseelement)
+    -   [Element and Component classes](#element-and-component-classes)
+    -   [PreactBaseElement callbacks](#preactbaseelement-callbacks)
+    -   [AMP/Preact Bridge](#amp-preact-bridge)
+-   [Element styling](#element-styling)
+-   [Register element with AMP](#register-element-with-amp)
+-   [Actions and events](#actions-and-events)
+-   [Allowing proper validation](#allowing-proper-validation)
+-   [Performance considerations](#performance-considerations)
+    -   [Loading external resources](#loading-external-resources)
+-   [Layouts supported in your element](#layouts-supported-in-your-element)
+    -   [What layout should your element support?](#what-layout-should-your-element-support-)
+-   [Experiments](#experiments)
+    -   [Enabling and removing your experiment](#enabling-and-removing-your-experiment)
+-   [Documenting your element](#documenting-your-element)
+-   [Example of using your extension](#example-of-using-your-extension)
+-   [Updating build configs](#updating-build-configs)
+-   [Versioning](#versioning)
+-   [Using Storybook](#using-storybook)
+-   [Unit tests](#unit-tests)
+-   [Linting and formatting](#linting-and-formatting)
+-   [Type checking](#type-checking)
+-   [Example PRs](#example-prs)
+
+## Getting started
+
+This document describes how to create a new Bento AMP extension, which is one of the most common ways of adding a new feature to AMP.
+
+Before diving into the details on creating a new Bento AMP extension, please familiarize yourself with the [general process for contributing code and features to AMP](https://github.com/ampproject/amphtml/blob/master/contributing/contributing-code.md). Since you are adding a new extension you will likely need to follow the [process for making a significant change](https://github.com/ampproject/amphtml/blob/master/contributing/contributing-code.md#process-for-significant-changes), including filing an ["Intent to Implement" issue](https://go.amp.dev/i2i) and finding a guide before you start significant development.
```

Comment:
![image](https://user-images.githubusercontent.com/6694512/104211309-e9d9d600-5401-11eb-8f99-bb2e507fd7ba.png)

To clarify I meant two different links:
- https://go.amp.dev/contribute/code
- https://go.amp.dev/contribute/code#process-for-significant-changes